src/pokedex_screen.c

```diff
@@ -2541,3 +2546,827 @@ static bool8 sub_81052D0(u8 a0)
+{
+    int i;
+    u8 count;
+
+    count = gDexCategories[categoryNum].count;
+
+    for (i = 0; i < count; i++)
+        if (sub_8106838(categoryNum, i))
+            return 1;
+
+    return 0;
+}
+
+void sub_81068DC(u8 categoryNum, u8 pageNum)
+{
+    int i, count;
+    u16 species;
+
+    count = gDexCategories[categoryNum].page[pageNum].count;
+    gUnknown_203ACF0->field_2C = 0;
+
+    for (i = 0; i < 4; i++)
+        gUnknown_203ACF0->field_18[i] = 0xffff;
+    for (i = 0; i < count; i++)
+    {
+        species = gDexCategories[categoryNum].page[pageNum].species[i];
+        if (sub_8106810(species) == TRUE)
+            if (sub_8104AB0(species, 0, 1))
+            {
+                gUnknown_203ACF0->field_18[gUnknown_203ACF0->field_2C]
+                    = gDexCategories[categoryNum].page[pageNum].species[i];
+                gUnknown_203ACF0->field_2C++;
+            }
+    }
+}
+
+u8 sub_810699C(u8 category)
+{
+    int i;
+    u8 count, v2, v3;
+
+    count = gDexCategories[category].count;
+    v2 = 0xff;
+    v3 = 0xff;
+
+    for (i = 0; i < count; i++)
+        if (sub_8106838(category, i))
+        {
+            if (v2 == 0xff)
+                v2 = i;
+            v3 = i;
+        }
+    if (v3 != 0xff)
+    {
+        gUnknown_203ACF0->field_29 = v2;
+        gUnknown_203ACF0->field_2A = v3 + 1;
+        return 0;
+    }
+    else
+    {
+        gUnknown_203ACF0->field_29 = 0;
+        gUnknown_203ACF0->field_2A = 0;
+        return 1;
+    }
+}
+
+u8 sub_8106A20(u16 a0)
+{
+    int i, j, k, categoryCount, categoryPageCount, v5;
+    u16 species;
+
+    for (i = 0; i < DEX_CATEGORY_SIZE; i++)
+    {
+        categoryCount = gDexCategories[i].count;
+        for (j = 0; j < categoryCount; j++)
+        {
+            categoryPageCount = gDexCategories[i].page[j].count;
+            for (k = 0, v5 = 0; k < categoryPageCount; k++)
+            {
+                species = gDexCategories[i].page[j].species[k];
+                if (a0 == species)
+                {
+                    gUnknown_203ACF0->field_28 = i;
+                    gUnknown_203ACF0->field_2B = j;
+                    gUnknown_203ACF0->field_2D = v5;
+                    return 0;
+                }
+                if (sub_8106810(species) == TRUE)
+                    if (sub_8104AB0(species, 0, 1))
+                        v5++;
+            }
+        }
+    }
+    return 1;
+}
+
+u8 sub_8106AF8(u16 a0)
+{
+    int i, v1;
+
+    for (i = 0, v1 = 0; i < a0; i++)
+        if (sub_8106838(gUnknown_203ACF0->field_28, i))
+            v1++;
+
+    return v1 + 1;
+}
+
+void sub_8106B34(void)
+{
+    if (JOY_NEW(START_BUTTON))
+        PlayCry2(gUnknown_203ACF0->field_5A, 0, 125, 10);
+}
+
+u8 sub_8106B60(u16 species)
+{
+    sub_8104AB0(species, 2, 1);
+    sub_8104AB0(species, 3, 1);
+
+    if (!IsNationalPokedexEnabled())
+        if (SpeciesToNationalPokedexNum(species) > KANTO_DEX_COUNT)
+            return CreateTask(sub_8106BD8, 0);
```

Comment:
Does `if (!IsNationalPokedexEnabled() && SpeciesToNationalPokedexNum(species) > KANTO_DEX_COUNT)` match?

Reply:
It does, changed